### ISSUE TITLE
Is a search term considered as "Information"? resulting in search failing 3.3.6: Error Prevention (All) 

### ISSUE THREAD

**Issue by jake-abma** [w3c/wcag#1755]
3.3.6: Error Prevention (All) talks about "submit information" in the SC text.

Now, if I use the search functionality, do I send "information" to a web page (or search engine provider!)?
I'd say they have information about my surf/search behavior.
(Or is the definition for WCAG another one?)

If it is information, you'll always fail a regular search as it will not be Reversible, Checked, or Confirmed.

The SC tries to fill another gap, I know, but that is not made clear in the SC text if 'search term' fits "information".

**Comment by jake-abma** [comment]
(or what does 'require' mean? Does Google 'require' you to submit information? Does a webshop 'require' you to order?)

**Comment by jake-abma** [comment]
(ps. 1.3.1: Info and Relationships  talks about "**Information**, structure, and relationships...")

**Comment by patrickhlauke** [comment]
For the question about what "require" means, also consider 3.3.2 Labels or Instructions "Labels or instructions are provided when content *requires* user input." (which you could then ask "so, are optional fields exempt from this?")

My personal view here is that the normative language is overly convoluted, when what it actually (I'm guessing here) means when saying "requires user input" / "require the user to submit information" is more along the lines of "allow users to enter information/make a selection in a form" - just a more generic/agnostic way of saying it, which unfortunately has some unintended connotations because of the meaning associated with "require" there.

Of course then this still leaves it open to interpretation what "submits information" means. Websites, search engines, etc gather lots of "information" even from the user just following a link (user agent, IP, cookies, etc). Plus, they then store that information for other uses (like building up your search history/profile) which most users may not even be aware of. Some search engines do allow you some level of going back in after the fact and removing some search history data (a la "don't use this for future shopping recommendations" type functionality), so in those cases the reversible aspect would be covered. confirmed would of course make little sense..."are you sure you want to start a search for X?". are search terms "checked" though, I wonder? generally, you can't (or it doesn't make sense to) submit an empty search. so there's at least a basic level of checking that usually happens on searches. you could then also argue that yes, it's checked if not empty, but as there is no such thing as a wrong search term, the check always comes back ok...a lawyery type answer here.

**Comment by patrickhlauke** [comment]
possibly along same lines, what about comments on a blog. if i mistype my name or message, but i can't then go back later to reverse my comment, does it fail? or does it count that the information was "checked" in the sense that the blog stopped me from posting a comment with an empty name, message, or something in the email field that doesn't look like an email address?

this SC can probably, under a certain light, be taken to fail a lot of things...

**Comment by alastc** [comment]
> If it is information, you'll always fail a regular search as it will not be Reversible, Checked, or Confirmed.

I think a search is reversible. I.e. you can change / reverse it. Nothing is saved or lost on submissions.

**Comment by patrickhlauke** [comment]
I'm guessing @jake-abma is referring to the information that the search engine stores ("this user searched for X") to build up its creepy profile about you so it can better serve you ads etc in future.

**Comment by jake-abma** [comment]
> I think a search is reversible. I.e. you can change / reverse it, you do another search. Nothing is saved or lost on submissions.

As @patrickhlauke mentioned, there's a twist in my question, see the last sentence in the opening message:

> The SC tries to fill another gap, I know, but that is not made clear in the SC text if 'search term' fits "information".

**Comment by patrickhlauke** [comment]
but as said, you could also take the interpretation to an extreme and say if i just follow a link/open a url, and the target site logs my visit (user agent, IP address, etc) in their logs, is that a (hidden, involuntary, un-opt-outable) "submission" of information? because then every site that uses even basic logging would fail here. but of course that's not the intent.

**Comment by jake-abma** [comment]
yeh, that's correct, the **"user to submit information"** part is just not specific enough in what is meant, following the WCAG 'we try to make it waterproof' rules. (we might need more definitions... :-) )
